Sources/MacroToolkit/Type.swift

```diff
@@ -174,6 +169,199 @@ public enum `Type`: TypeProtocol, SyntaxExpressibleByStringInterpolation {
+            attributedTypeSyntax?.baseType = TypeSyntax(memberTypeSyntax)
+            
+            return .member(.init(memberTypeSyntax, attributedSyntax: attributedTypeSyntax))
+            
+        case .metatype(let type):
+            let baseType = type._baseSyntax
+            let memberTypeSyntax = MemberTypeSyntax.init(
+                leadingTrivia: baseType.leadingTrivia,
+                baseType: baseType.baseType,
+                name: baseType.metatypeSpecifier,
+                trailingTrivia: baseType.trailingTrivia
+            )
+            if var attributedSyntax = type._attributedSyntax {
+                attributedSyntax.baseType = TypeSyntax(memberTypeSyntax)
+                return .member(.init(memberTypeSyntax, attributedSyntax: attributedSyntax))
+            } else {
+                return .member(.init(memberTypeSyntax))
+            }
+        case .missing(let type):
+            return .missing(.init(type._baseSyntax, attributedSyntax: type._attributedSyntax))
+        case .optional(let type):
+            var optionalTypeSyntax: OptionalTypeSyntax = type._baseSyntax
+            var attributedTypeSyntax: AttributedTypeSyntax? = type._attributedSyntax
+            let normalizedElement = Type(optionalTypeSyntax.wrappedType).normalized()
+            optionalTypeSyntax.wrappedType = TypeSyntax(normalizedElement._syntax)
+            attributedTypeSyntax?.baseType = TypeSyntax(optionalTypeSyntax)
+            
+//            let identifierSyntax = IdentifierTypeSyntax(
+//                leadingTrivia: optionalTypeSyntax.leadingTrivia,
+//                name: .identifier("Optional"),
+//                genericArgumentClause: .init(
+//                    arguments: .init(arrayLiteral: .init(argument: optionalTypeSyntax.wrappedType))
+//                ),
+//                trailingTrivia: optionalTypeSyntax.leadingTrivia
+//            )
+            
+            var base = "Optional<\(optionalTypeSyntax.wrappedType)>"
+            if let attributedTypeSyntax {
+                base = base.addingAttributes(from: attributedTypeSyntax)
+            }
+            return NormalizedType(stringLiteral: base)
+        case .packExpansion(let type):
+            // Looks like there can only be simple identifiers in pack expansions, with no generics, and therefore we
+            // don't ned to recursively normalize
+            
+            return .packExpansion(.init(type._baseSyntax, attributedSyntax: type._attributedSyntax))
+        case .packReference(let type):
+            // Looks like there can only be simple identifiers in pack references, with no generics, and therefore we
+            // don't ned to recursively normalize
+            
+            return .packReference(.init(type._baseSyntax, attributedSyntax: type._attributedSyntax))
+        case .simple(let type):
+            if type.name == "Void" {
+                return .tuple(.init(.init(elements: [])))
```

Comment:
I think `Void` is better to have as a simple type as that's the name most people know it by (and using `()` is considered less swift-y by a few core Swift contributors, even though `Void` is actually an alias to `()` under-the-hood).

Reply:
See https://github.com/stackotter/swift-macro-toolkit/pull/15#issuecomment-2241100094.